hardware_interface/CMakeLists.txt

```
@@ -40,8 +40,20 @@ target_include_directories(hardware_interface PUBLIC
   $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include/hardware_interface>
 )
-ament_target_dependencies(hardware_interface PUBLIC ${THIS_PACKAGE_INCLUDE_DEPENDS})
-
+target_link_libraries(hardware_interface PUBLIC
+  control_msgs::control_msgs
+  lifecycle_msgs::lifecycle_msgs
```

```suggestion
  {control_msgs_TARGETS}
  {lifecycle_msgs_TARGETS}
```
As per https://github.com/ros-controls/ros2_control/issues/2207